experimental/instrumentation-fetch/src/instrumentation.ts

```diff
@@ -0,0 +1,84 @@
+import { BaseInstrumentation, faro, globalObject } from '@grafana/faro-core';
+
+import { fetchGlobalObjectKey, originalFetch, originalFetchGlobalObjectKey } from './constants';
+
+export class FetchInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-sdk:instrumentation-fetch';
+  readonly version = '1.0.0'; // TODO - pull from package.json
+
+  initialize(): void {
+    console.log('Initializing FetchInstrumentation');
+    this.enable();
+  }
+
+  constructor() {
+    super();
+  }
+
+  /**
+   * Instrument fetch with Faro
+   */
+  private _instrumentFetch(): typeof fetch {
+    return function instrumentedFetch(input, init): Promise<Response> {
+      let copyInit = init ? Object.assign({}, init) : {};
+
+      let body;
+      if (copyInit && copyInit.body) {
+        body = copyInit.body;
+        copyInit.body = undefined;
+      }
+
```

Comment:
Suggestion: don't forget to add the `skipUrls` functionality. 
It's also important to prevent Faro from reporting on itself.

Reply:
Good point - this would require changes to the receiver? Or does this just mean generating some kind of unique request ID that we can then use in the backend? Depending on the scope I could take care of that with this PR.

Reply:
> Good point - this would require changes to the receiver? Or does this just mean generating some kind of unique request ID that we can then use in the backend? Depending on the scope I could take care of that with this PR.

I don't know yet but I expect we need changes on the receivers side. So imho it would make sense to tackle this as an own task because it may be a bigger thing. Also not sure if we need this for GA or not.